net: l2: ethernet: check dev->api and get_capabilities for being NULL

panekmaciej opened the pull request:
Adds missing checks for get_capabilities method not being NULL. Fixes crash with netusb and possibly other drivers.
See #77838

panekmaciej committed:
net: l2: ethernet: check if the dev->api->get_capabilities method not NULL

Adds missing checks for get_capabilities method not being NULL.
Fixes crash with netusb and possibly other drivers.

Signed-off-by: Maciej Panek <panekmaciej@outlook.com>

jukkar commented on subsys/net/l2/ethernet/ethernet.c:
```diff
@@ -207,29 +204,31 @@ static void ethernet_mcast_monitor_cb(struct net_if *iface, const struct net_add
+	dev = net_if_get_device(iface);
+	api = dev->api;
+
+	if (api && api->set_config) {
```
This change is for set_config callback, could you have a separate commit for it.

panekmaciej replied:
Done

jukkar commented on subsys/net/l2/ethernet/ethernet.c:
```diff
@@ -616,7 +615,7 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	struct net_pkt *orig_pkt = pkt;
 
-	if (!api) {
+	if (!api || !api->send) {
```
And this could be placed to separate commit too

jukkar reviewed (commented):
Looks good but the commit should split to three parts as you are also changing two other callbacks.

panekmaciej committed:
net: l2: ethernet: check if the dev->api->send method not NULL

adds a check for "send" method in Ethernet driver's API
not being NULL in ethernet_send

Signed-off-by: Maciej Panek <panekmaciej@outlook.com>

panekmaciej committed:
net: l2: ethernet: check if the dev->api not NULL

adds a check for Ethernet driver's API not being NULL
in ethernet_mcast_monitor_cb

Signed-off-by: Maciej Panek <panekmaciej@outlook.com>
